src/wp-includes/class-wp-block.php

```diff
@@ -607,3 +571,121 @@ public function render( $options = array() ) {
 		return $block_content;
 	}
 }
+
+// phpcs:disable Generic.Files.OneObjectStructurePerFile.MultipleFound
+/**
+ * HTML API Processor subclass implementing experimental set_inner_html.
+ *
+ * DO NOT USE THIS CLASS. Internal usage only.
+ *
+ * @access private
+ */
+class PrivateProcessor extends WP_HTML_Processor {
```

Comment:
https://github.com/WordPress/wordpress-develop/pull/6838 uses a pattern for a private processor class:

https://github.com/WordPress/wordpress-develop/blob/825f78effbbe234ef70a3502eeaad78ee833436c/src/wp-includes/class-wp-block.php#L338-L341